tests/compilation/test_helper_rl.py

```
@@ -77,6 +82,29 @@ def test_bqskit_to_qiskit_converts_u1q_to_r_gate() -> None:
     assert qc.data[0].operation.params == [0.1, 0.2]
 
 
+def test_bqskit_mapping_factory_flattens_workflows(monkeypatch: pytest.MonkeyPatch) -> None:
+    """Test that a workflow passed to the mapping factory is flattened."""
+    captured_workflow: Workflow | None = None
+
+    def fake_run(circuit: Circuit, workflow: Workflow, request_data: bool = False) -> tuple[Circuit, SimpleNamespace]:
```

Ruff (0.16.1)

[warning] Boolean-typed positional argument in function definition

(FBT001)